### PR TITLE
Tsmith/fix auth

### PR DESCRIPTION
**Pull request by Bento007**
#### Reviewers
**Functional:** @tihuan 

**Readability:** @MDunitz 

---

## Changes
- Fix for #981 
- modify samesite to lax for session cookie.
- Using Connexion problem exceptions.
- Removing unused authn code.

### CODE REVIEW

File: tests/unit/backend/chalice/api_server/test_v1_auth.py [hunk 1]
```diff
@@ -45,15 +45,17 @@ def test__auth_flow(self):
...
 
             # follow redirect
             test_url = f"/dp/v1/oauth2/callback?code=fakecode&state={args['state']}"
             response = self.app.get(test_url, headers=dict(host="localhost", Cookie=response.headers["Set-Cookie"]))
             self.assertEqual(response.status_code, 302)
             self.assertEqual(response.headers["Location"], self.auth_config.redirect_to_frontend)
             self.assertTrue("Set-Cookie" in response.headers)
-            self.check_set_cookie_is_secure(response.headers["Set-Cookie"])
+            self.check_set_cookie_is_secure(
+                response.headers["Set-Cookie"], expected_flags=["HttpOnly", "Secure", "SameSite=Strict"]
```
Comment (tihuan) [inline tests/unit/backend/chalice/api_server/test_v1_auth.py:57]: is this supposed to be `SameSite=Strict` or `Lax`? Thank you!

**Comment by tihuan** [comment]
I'm actually able to log in with Strict cookies. I used email to log in 😄 

demo

File: tests/unit/backend/chalice/api_server/test_v1_auth.py (hunk 1 above)
Comment (Bento007) [reply tests/unit/backend/chalice/api_server/test_v1_auth.py:57]: this cookie should be `Strict`

Comment (tihuan) [reply tests/unit/backend/chalice/api_server/test_v1_auth.py:57]: Got it thank you!

File: tests/unit/backend/chalice/api_server/test_v1_auth.py [hunk 2]
```diff
@@ -45,15 +45,17 @@ def test__auth_flow(self):
             self.assertEqual(args["client_id"], self.auth_config.client_id)
             self.assertEqual(args["response_type"], "code")
             self.assertTrue("/dp/v1/oauth2/callback" in args["redirect_uri"])
-            self.check_set_cookie_is_secure(response.headers["Set-Cookie"])
+            self.check_set_cookie_is_secure(response.headers["Set-Cookie"], ["HttpOnly", "Secure", "SameSite=Lax"])
```
Comment (tihuan) [inline tests/unit/backend/chalice/api_server/test_v1_auth.py:48]: Can we add comment that this is for asserting `session` cookie and the next block is for `cxguser` cookie please? At a glance, it's hard to tell which cookie we're asserting 😄 Thank you!

**Review by tihuan — APPROVED** [review]
One tiny comment otherwise LGTM! Thanks for catching and fixing this, @Bento007 🙏

**Review by MDunitz — APPROVED** [review]
lgtm

File: tests/unit/backend/chalice/api_server/test_v1_auth.py (hunk 2 above)
Comment (Bento007) [reply tests/unit/backend/chalice/api_server/test_v1_auth.py:49]: Good idea. It is confusing. Done!

**Pull request merged**